homeassistant/components/orangepi_gpio/switch.py

```diff
@@ -0,0 +1,86 @@
+"""Allows to configure a switch using Orange Pi GPIO."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.components import orangepi_gpio
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['orangepi_gpio']
+
+CONF_PORTS = 'ports'
+CONF_PINMODE = 'pinmode'
+CONF_INVERT_LOGIC = 'invert_logic'
+
+PINMODES = ['pc', 'zeroplus', 'zeroplus2', 'deo', 'neocore2']
+
+DEFAULT_INVERT_LOGIC = False
+
+_SWITCHES_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORTS): _SWITCHES_SCHEMA,
+    vol.Required(CONF_PINMODE): vol.In(PINMODES),
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Orange Pi GPIO devices."""
+    pinmode = config.get('pinmode')
```

Comment:
Use the constants defined
```suggestion
    pinmode = config[CONF_PINMODE]
```